core/basic_authorship/impl/block_builder_factory_impl.cpp

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "basic_authorship/impl/block_builder_factory_impl.hpp"
+#include "basic_authorship/impl/block_builder_impl.hpp"
+
+namespace kagome::basic_authorship {
+
+  BlockBuilderFactoryImpl::BlockBuilderFactoryImpl(
+      std::shared_ptr<runtime::Core> r_core,
+      std::shared_ptr<runtime::BlockBuilderApi> r_block_builder,
+      std::shared_ptr<blockchain::HeaderBackend> header_backend)
+      : r_core_(std::move(r_core)),
+        r_block_builder_(std::move(r_block_builder)),
+        header_backend_(std::move(header_backend)) {}
```

Comment:
Recently we have decided that adding `BOOST_ASSERT`-s for params, which are nullable, is a good idea. So, please, add them for `r_core_` and other variables in this and other added constructors